MekHQ/src/mekhq/campaign/rating/AbstractUnitRating.java

```diff
@@ -498,48 +502,26 @@ protected void initValues() {
         getLogger().methodEnd(getClass(), "initValues()");
     }
 
-    private void updateBayCount(Dropship ds) {
-        // ToDo Superheavy Bays.
-        for (Bay bay : ds.getTransportBays()) {
-            if (bay instanceof MechBay) {
-                setMechBayCount(getMechBayCount() + (int)bay.getCapacity());
-            } else if (bay instanceof BattleArmorBay) {
-                setBaBayCount(getBaBayCount() + (int)bay.getCapacity());
-            } else if (bay instanceof InfantryBay) {
-                setInfantryBayCount(getInfantryBayCount() +
-                                    (int) (bay.getCapacity() / ((InfantryBay) bay).getPlatoonType().getWeight()));
-            } else if (bay instanceof LightVehicleBay) {
-                setLightVeeBayCount(getLightVeeBayCount() +
-                                    (int) bay.getCapacity());
-            } else if (bay instanceof HeavyVehicleBay) {
-                setHeavyVeeBayCount(getHeavyVeeBayCount() +
-                                    (int) bay.getCapacity());
-            } else if (bay instanceof ASFBay) {
-                setFighterBayCount(getFighterBayCount() +
-                                   (int) bay.getCapacity());
-            } else if (bay instanceof SmallCraftBay) {
-                setSmallCraftBayCount(getSmallCraftBayCount() +
-                                      (int) bay.getCapacity());
-            }
-        }
-    }
-
     void updateBayCount(Entity e) {
-        if (e instanceof Dropship) {
-            updateBayCount((Dropship) e);
-        } else if (e instanceof Jumpship) {
-            updateBayCount((Jumpship) e);
-        }
-    }
-
-    private void updateBayCount(Jumpship jumpship) {
-        for (Bay bay : jumpship.getTransportBays()) {
-            if (bay instanceof ASFBay) {
-                setFighterBayCount(getFighterBayCount() +
-                                   (int) bay.getCapacity());
-            } else if (bay instanceof SmallCraftBay) {
-                setSmallCraftBayCount(getSmallCraftBayCount() +
-                                      (int) bay.getCapacity());
+        if ((e instanceof Jumpship) || (e instanceof SmallCraft)) {
```

Comment:
`Entity::getTransportBays` always returns a non-null Vector. I wonder if this `if` test can be dropped to simplify the code (and perhaps future proof it). Thoughts @neoancient @mkerensky @NickAragua ?

Reply:
Unless I missed something, this count is to determine the unit capacity for interstellar transport, so it should only be checking Jumpships, Warships, and Dropships. While small craft mech/aero/vehicle bays are impractical, they can have infantry bays that should not count toward transport, since the infantry are not going to be transported in an infantry bay of a small craft inside a small craft bay. In the near future when large naval and airship support vehicles are implemented, it would also be possible for them to have a variety of bays that also should not count toward transport capacity. `if (e.isLargeCraft())` would include dropships, jumpships, and warships. But it would also include space stations, which also shouldn't count toward transport capacity.

Reply:
You could easily have people living in small craft bays, although I wouldn't recommend it. Is there be a way to put a morale penalty on the unit inside if they live there long-term?
Also, no matter how this is programmed Space Stations will be included, and there is no easily way that I've found to remove them from the count.

Reply:
And also, removing space stations from transport capacity opens up a whole bag of worms. First of all, why does a mercenary company have a space station? Is it because it is a merc company base? In that case, we definitely need to count it, for base requirements (given how you can easily simulate having a base with the splitting force mechanics) should show any bays the company has.

Reply:
I was simply responding to @sixlettervariables question about whether I thought the if statement was necessary, and I believe it is. I don't intend to get pulled into a debate about how it should work. But in response to your objections:
1. You can suggest all sorts of ways to transport infantry in a make-shift way. This should be kept simple. If you want to claim that infantry can live in a bay in a docked small craft you would then need to make sure the unit has transport capacity for that small craft. That's an unnecessary complication.
2. Space stations can easily be excluded from the bay count in the if statement in question.
3. MekHQ works for any force, not just mercenaries. FMM is designed for merc units but the CO rating is for any unit.
4. Transport capacity is about the ability to move forces between systems, not the ability to house them. Space stations do not contribute to transport ability.

Reply:
My apologies, was spitballing ideas and forgot that you aren't used to how I tend to operate.
For 1, this isn't a makeshift bay. This is a bay that is the same size and with the same capacity as that on a dropship. I was thinking a morale penalty might be a nice way to show that the extra space is a bit cramped, but... RAW they are the same size and ability as the dropship ones.
For the rest, was just trying to throw out ideas, things to think about and consider.

Reply:
Thank you for the clarification @neoancient

Reply:
This still includes small craft and space station bays. FMM(r), p. 154 and Campaign Operations, p. 35 both state explicitly that transport capacity is based on dropships, jumpships, and warships.

Reply:
I've made those changes. That means that #1100, which started this, is not a bug at all. However, I've made some internal changes to allow for a new unit rating that decides its bay count separately from the main two, and there are some other fixes in this PR. So, I think this PR should go in, but renamed (probably to something like Unit Rating: Minor bug fixes).

Also, I am considering making a new type of rating to handle stationary forces or single system forces, but that's future ideas/plans.

Reply:
> 
> 
> This still includes small craft and space station bays. FMM(r), p. 154 and Campaign Operations, p. 35 both state explicitly that transport capacity is based on dropships, jumpships, and warships.

And neither actually excludes sub-bays on the larger vessels.  You are the one making an arbitrary decision on what equipment is and is not valid.